Sources/NIOPosix/PendingWritesManager.swift

```diff
@@ -434,9 +434,10 @@ final class PendingStreamWritesManager: PendingWritesManager {
         case .fileRegion(let file):
             let readerIndex = file.readerIndex
             let endIndex = file.endIndex
-            return try file.fileHandle.withUnsafeFileDescriptor { fd in
-                self.didWrite(itemCount: 1, result: try operation(fd, readerIndex, endIndex))
+            let writeResult = try file.fileHandle.withUnsafeFileDescriptor { fd in
+                try operation(fd, readerIndex, endIndex)
             }
+            return self.didWrite(itemCount: 1, result: writeResult)
```

Comment:
This was always a bug: Should've never called out whilst in `withUnsafeFileDescriptor`...